src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueDebtAdaptor.sol

```diff
@@ -0,0 +1,244 @@
+        return ERC20(market.loanToken);
+    }
+
+    /**
+     * @notice This adaptor reports values in terms of debt.
+     * @return Whether or not this adaptor is in terms of debt.
+     */
+    function isDebt() public pure override returns (bool) {
+        return true;
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    // `borrowAsset`
+    /**
+     * @notice Allows strategists to borrow assets from Morpho Blue.
+     * @param _id encoded bytes32 MB id that represents the MB market for this position.
+     * @param _amountToBorrow the amount of `loanToken` to borrow on the specified MB market.
+     */
+    function borrowFromMorphoBlue(Id _id, uint256 _amountToBorrow) public {
+        _validateMBMarket(_id);
+        MarketParams memory market = morphoBlue.idToMarketParams(_id);
+        _borrowAsset(market, _amountToBorrow, address(this));
+
+        // Check if borrower is insolvent (AKA they have bad LTV), revert if they are
+        if (minimumHealthFactor > (_getHealthFactor(_id, market))) {
+            revert MorphoBlueDebtAdaptor__HealthFactorTooLow(_id);
+        }
+    }
+
+    // `repayDebt`
+
+    /**
+     * @notice Allows strategists to repay loan debt on Morph Blue Lending Market. Make sure to call addInterest() beforehand to ensure we are repaying what is required.
+     * @dev Uses `_maxAvailable` helper function, see BaseAdaptor.sol
+     * @param _id Encoded bytes32 MB id that represents the MB market for this position.
+     * @param _debtTokenRepayAmount The amount of `loanToken` to repay.
+     */
+    function repayMorphoBlueDebt(Id _id, uint256 _debtTokenRepayAmount) public {
+        _validateMBMarket(_id);
+        MarketParams memory market = morphoBlue.idToMarketParams(_id);
+        _accrueInterest(market); // either call this or use periphery library.
+
+        ERC20 tokenToRepay = ERC20(market.loanToken);
+        uint256 debtAmountToRepay = _maxAvailable(tokenToRepay, _debtTokenRepayAmount);
+
+        // using Morpho sharesLibrary we can calculate the sharesToRepay from the debtAmount
+        uint256 totalBorrowAssets = morphoBlue.market(_id).totalBorrowAssets;
+        uint256 totalBorrowShares = morphoBlue.market(_id).totalBorrowShares;
+        uint256 sharesToRepay = debtAmountToRepay.toSharesUp(totalBorrowAssets, totalBorrowShares); // get the total assets and total borrow shares of the market
+
+        // MorphoBlue reverts w/ underflow/overflow error if trying to repay with more than we have. That said, we will manage when strategists try to pass in type(uint256).max
+        uint256 sharesAccToMorphoBlue = morphoBlue.borrowShares(_id, address(this));
+        if (sharesAccToMorphoBlue < sharesToRepay) {
+            sharesToRepay = sharesAccToMorphoBlue;
+        }
+        
+        tokenToRepay.safeApprove(address(morphoBlue), type(uint256).max);
```

Comment:
Wouldn't the amount be `debtAmountToRepay` instead of a `type(uint256).max`? If there is weird rounding errors, then its fine to use the uint256 max logic since we revoke approval afterwards, but if we do know the exact amount of assets morpho needs to repay the debt, then we should use that.

Reply:
Yeah it would be `debtAmountToRepay` if `sharesAccToMorphoBlue > sharesToRepay `. Either way though, approving what the cellar actually has is the most that can be done. I'll change it to `debtAmountToRepay`